Leanplum-SDK/Classes/Internal/Leanplum.m

```diff
@@ -1834,20 +1834,26 @@ + (void)maybePerformActions:(NSArray *)whenConditions
     // Make sure to capture the held back.
     [LPActionContext sortByPriority:actionContexts];
     NSNumber *priorityThreshold = [((LPActionContext *) [actionContexts firstObject]) priority];
-    BOOL messageActionTriggered = NO;
-    
+    NSNumber *countDownThreshold = [self fetchCountDownForContext: (LPActionContext *) [actionContexts firstObject]];
+    BOOL isPrioritySame = NO;
     for (LPActionContext *actionContext in actionContexts) {
         NSNumber *priority = [actionContext priority];
         if (priority.intValue > priorityThreshold.intValue) {
             break;
         }
-        
+        if (isPrioritySame) {//priority is same
```

Comment:
Can you elaborate?

Reply:
Yes, thats what i wanted as well ideally.  Any ideas how we can add these test cases? I had thought of it. We might be able to add XCUITests for these, but we need to strategize them and  in different PR. That will happen on Rondo iOS.
 There are 2 problems with current code which i found.
1. As per my understanding the api's are not exposed for knowing what is scheduled or not.
2. Even if its possble, we will have to have 2 different json files with different Message countdowns. With current spagetti code, it might be difficult.